File: design-patterns/go/builder/vehicle/vehicle_builder.go
// vehicle_builder.go

package main

type VehicleBuilder interface {
	AddDoor(noOfDoor int)
	AddEngine(noOfEngine int)
	AddInterior()
	AddSeat(noOfSeat int)
	AddWheel(noOfWheel int)
	AddWing(noOfWing int)
}

File: design-patterns/ts/builder/vehicle/vehicle-builder.ts
// vehicle-builder.ts

interface VehicleBuilder {
    addWheel(noOfWheel:number): void;
    addEngine(noOfEngine:number): void;
    addSeat(noOfSeat:number): void;
    addInterior(): void;
    addDoor(noOfDoor:number): void;
    addWing(noOfWing:number): void;
}

export default VehicleBuilder;